src/client.js

```diff
@@ -286,34 +285,40 @@ Client.prototype.connect = function(args) {
 	client.save();
 };
 
-Client.prototype.updateToken = function(callback) {
-	var client = this;
-
-	crypto.randomBytes(48, function(err, buf) {
+Client.prototype.generateToken = function(callback) {
+	crypto.randomBytes(48, (err, buf) => {
 		if (err) {
 			throw err;
 		}
 
-		callback(client.config.token = buf.toString("hex"));
+		callback(buf.toString("hex"));
 	});
 };
 
+Client.prototype.updateSession = function(token, ip, request) {
+	const client = this;
+	const agent = UAParser(request.headers["user-agent"] || "");
+
+	client.config.sessions[token] = {
+		lastUse: Date.now(),
+		ip: ip,
+		agent: `${agent.browser.name} ${agent.browser.major} on ${agent.os.name} ${agent.os.version}`,
```

Comment:
It's either storing full raw user agent, or the pre-formatted one. I opted for the formatted one.

I do want to store user agent info in this PR because I don't know how long it will take for the UI to ship, and I don't want to miss this info.